### PR TITLE
[fix](function) fix error result in split_by_string with utf8 chars

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

```
mysql []>select split_by_string('你a好b世c界','');
+-----------------------------------------------------------------------------+
| split_by_string('你a好b世c界', '')                                          |
+-----------------------------------------------------------------------------+
| ["�", "�", "�", "a", "�", "�", "�", "b", "�", "�", "�", "c", "�", "�", "�"]             |
+-----------------------------------------------------------------------------+
1 row in set (0.03 sec)
now
mysql []>select split_by_string('你a好b世c界','');
+---------------------------------------------+
| split_by_string('你a好b世c界', '')          |
+---------------------------------------------+
| ["你", "a", "好", "b", "世", "c", "界"]     |
+---------------------------------------------+
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -2251,6 +2221,34 @@ class FunctionSplitByString : public IFunction {
...
+
+    void split_empty_delimiter(const StringRef& str_ref, ColumnString::Chars& column_string_chars,
+                               ColumnString::Offsets& column_string_offsets,
+                               NullMapType* dest_nested_null_map, ColumnArray::Offset64& string_pos,
+                               ColumnArray::Offset64& dest_pos) const {
+        const size_t old_size = column_string_chars.size();
+        const size_t new_size = old_size + str_ref.size;
+        column_string_chars.resize(new_size);
+        memcpy(column_string_chars.data() + old_size, str_ref.data, str_ref.size);
+        if (simd::VStringFunctions::is_ascii(str_ref)) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:2233]: better `std::iota + resize` do faster work

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**